src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.cs

```diff
@@ -189,38 +203,58 @@ public static Vector4 Atop(Vector4 dst, Vector4 src, Vector4 blend)
             return color;
         }
 
+        /// <summary>
+        /// Returns the result of the "In" compositing equation.
+        /// </summary>
+        /// <param name="destination">The destination vector.</param>
+        /// <param name="source">The source vector.</param>
+        /// <param name="blend">The amount to blend. Range 0..1</param>
+        /// <returns>The <see cref="Vector4"/>.</returns>
         [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        public static Vector4 In(Vector4 dst, Vector4 src, Vector4 blend)
+        public static Vector4 In(Vector4 destination, Vector4 source, Vector4 blend)
         {
-            float alpha = dst.W * src.W;
+            // TODO: blend is not used?
+            float alpha = destination.W * source.W;
```

Comment:
This concerns me. We have generated many methods that call this and I would expect it to use the blend vector and yield a different response. I'll need help implementing this.

Reply:
Just an assumption, but I think `blend` is only a parameter to simplify method generators by using uniform call sites. (Compare `AddSrcOver` vs `AddSrcIn` in `PorterDuffFunctions.Generated.cs`.

I need more research to be 100% sure though. Maybe @vpenades or @tocsoft can confirm?

Reply:
The original Porter Duff formulas are based on a table where all the terms are multipled either by Zero, One or Blend.   There's a few formulas that don't use Blend or it's simply cancelled out, in these cases, the _Source_ is not actually blended with _Dest_ , but is used as a mask. __Xor__ is a good example.

@antonfirsov is right, the Blend parameter is there to simplify code generation, even if some code paths don't use it. The other option would be to hand write all the methods.

Reply:
I'ts been a while since I visited that code.... but I can take a deeper look, let me know if you need some clarification about some specific method.

Reply:
@vpenades Thanks for getting back to us, very much appreciated!

This makes sense then, I think I've found the table you were referencing.

http://ssp.impulsetrain.com/porterduff.html

Reply:
@vpenades  The only question then I have is with the `In` method. I can see that the template calls the Blend method for each type but the function implementation never uses the result. 

https://github.com/SixLabors/ImageSharp/blob/e484a405dfe8376a55f68cba96129c8c5522c0ff/src/ImageSharp/PixelFormats/PixelBlenders/PorterDuffFunctions.Generated.tt#L61

It looks like I could simplify the code here and remove the overhead of generating the blend result to pass to the Porter-Duff function.

Reply:
Yes, the code generation can be optimized even further to avoid calling the blend when it's not needed.

At the time I wrote that code I didn't optimize it further because the code was becoming complicated enough, and because I wanted to leave a sense of completeness, in case we wanted to add more composition modes.

Funny thing is, the official [W3 composition specification](https://www.w3.org/TR/compositing-1/#security) states that all composition modes should have the same CPU cost to avoid a timing attack.

In a software implementation it would mean making all composition modes as expensive as the most expensive mode, which is certainly not practical for most use cases.

Reply:
Perfect! I've done a bit of cleanup then to remove the unnecessary calculation.

I read the specification you listed and I think we're ok. Your implementation is pixel agnostic anyway.

Thanks again for your help, very much appreciated! 👍